tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/EmailAddressContact.scala

```diff
@@ -0,0 +1,39 @@
+package com.linagora.tmail.james.jmap.model
+
+import com.google.common.collect.{HashMultimap, Multimap, Multimaps}
+import org.apache.james.jmap.api.model.AccountId
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.util.UUID
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+case class EmailAddressContact (id: UUID, address: String) {
+  def contains(part: String): Boolean = address.contains(part)
+}
+
+trait EmailAddressContactSearchEngine {
+  def index(accountId: AccountId, contact: EmailAddressContact): Publisher[Unit]
+
+  def autoComplete(accountId: AccountId, part: String): Publisher[EmailAddressContact]
+
+}
+
+class InMemoryEmailAddressContactSearchEngine extends EmailAddressContactSearchEngine {
+  val emailList: Multimap[AccountId, EmailAddressContact] = Multimaps.synchronizedSetMultimap(HashMultimap.create())
+
+  override def index(accountId: AccountId, address: EmailAddressContact): Publisher[Unit] = {
```

Comment:
Benoit always remind me to remove unnecessary bracket `{}` when the method is short

Reply:
Done fix, looks much better, thanks, Tung :D